internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1450,6 +1450,21 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 
 			"tags": commonschema.Tags(),
 
+			"upgrade_override_setting": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"effective_until": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.IsRFC3339Time,
+						},
+					},
+				},
+			},
```

Comment:
Is this the only configurable property in this block? If so, it might be worth flattening this down to a single property, for example `upgrade_override_until`?

What happens to this value when the specified point in time passes, is it removed from the response, or still returned?  

Should we also validate that the supplied time is in the future?